lua/lspconfig/server_configurations/drools_lsp.lua

```diff
@@ -1,35 +1,60 @@
 local util = require 'lspconfig.util'
 
-local get = function(config, section, key)
-  return vim.tbl_get(config.settings, section, key) and config.settings[section][key] or nil
+local function get(config, segments)
+  if vim.tbl_get(config, unpack(segments)) then
+    local cfg = config
+    for _, seg in ipairs(segments) do
+      cfg = cfg[seg]
+    end
+    return cfg
+  end
+  return nil
 end
 
-local get_java_bin = function()
-  local java_bin = vim.env.JAVA_HOME and util.path.join(vim.env.JAVA_HOME, 'bin', 'java') or 'java'
-  if vim.fn.has 'win32' == 1 then
-    java_bin = java_bin .. '.exe'
+local function get_java_bin(config)
+  local java_bin = get(config, { 'drools', 'java', 'bin' })
+  if java_bin == nil then
+    java_bin = vim.env.JAVA_HOME and util.path.join(vim.env.JAVA_HOME, 'bin', 'java') or 'java'
+    if vim.fn.has 'win32' == 1 then
+      java_bin = java_bin .. '.exe'
+    end
   end
   return java_bin
 end
 
+local function get_java_opts(config)
+  local java_opts = get(config, { 'drools', 'java', 'opts' })
+  return java_opts and java_opts or {}
+end
+
+local function get_jar(config)
+  local drools_jar = get(config, { 'drools', 'jar' })
+  if drools_jar == nil then
+    drools_jar = 'drools-lsp-server-jar-with-dependencies.jar'
+  end
+  return drools_jar
+end
+
+local function get_cmd(config)
+  local cmd = get(config, { 'cmd' })
+  if cmd == nil then
```

Comment:
```suggestion
  if vim.tbl_isempty(cmd) then
```

Reply:
how do you think this change :)

Reply:
@glebnir, juts like java_bin, `cmd` is also a string, not a table

Reply:
@glepnir Oops, my last comment on this line was wrong, so changing it. :) Sure, I can make this change.

Reply:
I prefer to check for nil but not check for empty table. It will only be empty if the user set it and made it empty. At which point, it's their fault they did that.

Reply:
the nil check can be simply like `if somethint then` `if not something then`.

Reply:
Okay. Will change.